tin/apps/assignments/models.py

```diff
@@ -194,41 +194,53 @@ def save_grader_file(self, grader_text: str) -> None:
             logger.error("Cannot run processes: %s", e)
             raise FileNotFoundError from e
 
-    def list_files(self) -> list[tuple[int, str, str, int, datetime.datetime]]:
+    def list_files(self) -> list[tuple[int, Path, int, datetime.datetime]]:
+        """List out every file in an assignment
+
+        This excludes grader files and grader log files.
+
+        Returns:
+            A list where each element is a tuple of
+
+            * The file index
+            * The file path
+            * The size of the file
+            * The files last modification time
+        """
         self.make_assignment_dir()
 
-        assignment_path = os.path.join(settings.MEDIA_ROOT, f"assignment-{self.id}")
+        assignment_path = settings.MEDIA_ROOT / f"assignment-{self.id}"
 
         files = []
         grader_file = upload_grader_file_path(self, "")
         grader_log_file = self.grader_log_filename
 
-        for i, item in enumerate(os.scandir(assignment_path)):
-            if item.is_file():
-                stat = item.stat(follow_symlinks=False)
+        for i, path in enumerate(assignment_path.iterdir()):
+            if path.is_file():
+                stat = path.stat(follow_symlinks=False)
                 item_details = (
                     i,
-                    item.name,
-                    item.path,
+                    path,
```

Comment:
This change requires a change to `templates/assignments/manage_files.html`.

Reply:
Nice catch!